worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -111,7 +112,28 @@ class ConnectionProvider(props: Properties) {
       val newIP: String = getIP
       if (hasIpAddressChanged(pool, newIP)) {
         // A failover has occurred, so we evict connections softly. New connectons look up the new IP address
-        pool.connectionProvider.map(_.getDataSource().getHikariPoolMXBean().softEvictConnections())
```

Comment:
I have a theory that `softEvictConnections()` does not permit future connections be opened within the same pool. Creating a new HikariPool from scratch will guarantee to use the new IP address returned from the CNAME URL

Reply:
Possibly we are running into https://github.com/brettwooldridge/HikariCP/issues/1836 ?

> I have a theory that softEvictConnections() does not permit future connections be opened within the same pool.

I don't thilnk that is right. Nothing in the documentation indicates that behavior.

However, https://github.com/brettwooldridge/HikariCP/issues/228 does recommend replacing the entire pool, so I guess that might be a better approach.